docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,692 @@
+field. You specify a datatype and name for your columns, and then as data gets entered, rows (the long left/right) ones 
+get created, read, updated, and deleted (remember CRUD). One thing to note is that in relational databases, unlike 
+Excel, you don't have an unlimited amount of columns that can be automatically added. When you create your table, it has 
+a set structure, meaning you can only add/remove/update columns by changing the structure of the table. You can, 
+however, add an unlimited number of rows in theory, though you'll run out of space and your queries will start to slow 
+down after a while. 
+
+Here's a reminder of what an Excel sheet can look like:
+
+| These | are | your | columns | :) | 
+| :---: | :---: | :---: | :---: | :---: |
+| T | h | e | s | e |
+| a | r | e | :) | :) |
+| y | o | u | r | :) |
+| r | o | w | s | :) |
+
+### Data Types in SQL
+
+In SQL, when making new columns on a table, altering columns, and dealing with table data, you need to be aware of the
+data types of the variables you're working with. It's somewhat similar to working with types in Java, however, you don't
+need to declare types all over the place. It matters most when you're trying to perform operations on types and getting 
+the data using the JDBC/ODBC. For example, if you try to perform a query where you're adding two strings, you'll receive 
+an error since strings can't be added. You can concatenate two strings, but addition is for number types. 
+
+There are many types available, especially in PostgreSQL, but below are a few of the most commonly used ones.
+
+#### SERIAL
+
+The `SERIAL` type is an `INTEGER` under the hood, usually used as an ID. It's a 
+PostgreSQL specific data type, which is usually just an `INTEGER` with the `AUTO INCREMENT` modifier in most other 
+databases. This is really useful for integer-typed
+`PRIMARY KEYS` in other databases, since it automatically creates a new unique ID for you.
+
+#### CHAR
+
+The `CHAR(x)` type is actually a string of max length `x`. If a value you insert isn't the full length `x`, then the 
+value will be right-padded with spaces. For example, if the column type is `CHAR(5)`, then "hello" would be saved as 
+"hello", but "hey" would be saved as "hey__" (pretend the underscores are spaces).
+
+#### VARCHAR
+
+The `VARCHAR(x)` type is a variable-length string of max length `x`. If a value you insert isn't the full length `x`, 
+then the value will _not_ be right-padded with spaces. For example, if the column type is `VARCHAR(5)`, then "hello" 
+would be saved as "hello" and "hey" would be saved as "hey".
+
+
+##### CHAR vs VARCHAR
+
+The difference between `CHAR(x)` and `VARCHAR(x)` is that a `CHAR` will right-pad the entry with spaces if it isn't of 
+length `x`, but a `VARCHAR` would save the entered value. If either columns attempt to save a value that is longer than
+`x` characters, then the value will be truncated and an error or warning might be thrown.
+
```

Comment:
Maybe after this you can do like an interactive example with them and ask what they think the database is going to store when you give it a certain type and value?